src/constructors/regular.c

```diff
@@ -408,6 +408,104 @@ igraph_error_t igraph_tree(igraph_t *graph, igraph_integer_t n, igraph_integer_t
     return IGRAPH_SUCCESS;
 }
 
+/**
+ * \ingroup generators
+ * \function igraph_symmetric_tree
+ * \brief Creates a symmetric tree which has the same number of branchss at each level.
+ *
+ * \param graph Pointer to an uninitialized graph object.
+ * \param vector Vector detailing the number of branches at each level
+ * \param type Constant, gives whether to create a directed tree, and
+ *        if this is the case, also its orientation. Possible values:
+ *        \clist
+ *        \cli IGRAPH_TREE_OUT
+ *          directed tree, the edges point
+ *          from the parents to their children,
+ *        \cli IGRAPH_TREE_IN
+ *          directed tree, the edges point from
+ *          the children to their parents.
+ *        \cli IGRAPH_TREE_UNDIRECTED
+ *          undirected tree.
+ *        \endclist
+ * \return Error code:
+ *         \c IGRAPH_INVMODE: invalid mode argument.
+ *
+ * Time complexity: O(|V|+|E|), the
+ * number of vertices plus the number of edges in the graph.
+ *
+ * \sa \ref igraph_tree() \ref igraph_lattice(), \ref igraph_star() for creating other regular
+ * structures; \ref igraph_from_prufer() for creating arbitrary trees;
+ * \ref igraph_tree_game() for uniform random sampling of trees.
+ *
+ * \example examples/simple/igraph_symmetric_tree.c
+ */
+
+igraph_error_t igraph_symmetric_tree(igraph_t *graph, igraph_vector_int_t *dimvector,
+                igraph_tree_mode_t type) {
+    // TODO: implement function
+    igraph_vector_int_t edges = IGRAPH_VECTOR_NULL;
+    igraph_integer_t i, j, k, m;
+    igraph_integer_t idx = 0;
+    igraph_integer_t to = 1;
+    igraph_integer_t n = 1 + VECTOR(*dimvector)[0] + VECTOR(*dimvector)[0]*VECTOR(*dimvector)[1] + VECTOR(*dimvector)[0]*VECTOR(*dimvector)[1]*VECTOR(*dimvector)[2];
+
+    
+    if ( igraph_vector_int_empty(dimvector) || igraph_vector_int_min(dimvector) < 1) {
```

Comment:
An empty vector should result in a singleton graph, and is not an error.

Reply:
Thanks for your reviews. I adapted the proposed changes.